kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -124,13 +124,29 @@ protected BaseOperation(OkHttpClient client, Config config, String apiGroup, Str
    */
   private static <T> String apiVersion(T item, String apiVersion) {
     if (apiVersion != null && !apiVersion.isEmpty()) {
-      return apiVersion;
+      return trimVersion(apiVersion);
     } else if (item instanceof HasMetadata) {
-      return ((HasMetadata)item).getApiVersion();
+      return trimVersion(((HasMetadata)item).getApiVersion());
     }
     return null;
   }
 
+  /**
+   * Separates apiVersion for apiGroup/apiVersion combination.
+   * @param apiVersion  The apiVersion or apiGroup/apiVersion combo.
+   * @return            Just the apiVersion part without the apiGroup.
+     */
+  private static String trimVersion(String apiVersion) {
+    if (apiVersion == null) {
+      return null;
+    } else if (apiVersion.contains("/")) {
```

Comment:
If you don't check for contains("/") you'd need to check the length of the array afterwards.
So complexity wise its the same thing. Now performance wise, in the worst case (no / is found) both approach should behave the same. In all other cases its possibly cheaper to check the length of an array compare to check contains. And thank you so much for spotting it :-)